sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts

```diff
@@ -191,53 +228,57 @@ export class ConfirmationOfEnrollmentController {
    */
   @HasLocationAccess("locationId")
   @Patch(
-    ":locationId/confirmation-of-enrollment/application/:applicationId/confirm",
+    ":locationId/confirmation-of-enrollment/disbursement/:disbursementScheduleId/confirm",
   )
   async confirmEnrollment(
     @Param("locationId") locationId: number,
-    @Param("applicationId") applicationId: number,
+    @Param("disbursementScheduleId") disbursementScheduleId: number,
+    @UserToken() userToken: IUserToken,
   ): Promise<void> {
-    // get application for the location if COE status is `REQUIRED`
-    const application =
-      await this.applicationService.getApplicationDetailsForCOE(
+    // Get the disbursement and application summary for COE.
+    const disbursementSchedule =
+      await this.disbursementScheduleService.getDisbursementAndApplicationSummary(
         locationId,
-        applicationId,
+        disbursementScheduleId,
       );
-    if (!application) {
-      throw new NotFoundException("Application Not Found");
+
+    if (!disbursementSchedule) {
+      throw new NotFoundException(COE_NOT_FOUND_MESSAGE);
     }
     // institution user can only confirm COE, when the student is
-    // within COE_WINDOW of their Program Start date
+    // within COE_WINDOW of disbursement date
     if (
       !this.applicationService.withinValidCOEWindow(
-        application.offering.studyStartDate,
+        disbursementSchedule.disbursementDate,
       )
     ) {
       throw new UnprocessableEntityException(
         `Confirmation of Enrollment window is greater than ${COE_WINDOW} days`,
       );
     }
 
-    //TODO: Document Number should be Updated in the same transaction when COE status changes
-    await this.disbursementScheduleService.updateDisbursementScheduleDocumentNumber(
-      applicationId,
-    );
-
-    const updatedCOEStatus =
-      await this.applicationService.updateApplicationCOEStatus(
-        applicationId,
-        COEStatus.completed,
-        ApplicationStatus.completed,
+    const firstOutstandingDisbursement =
+      await this.disbursementScheduleService.getFirstOutstandingCOE(
+        disbursementSchedule.application.id,
       );
 
-    if (updatedCOEStatus.affected === 0) {
+    if (disbursementSchedule.id !== firstOutstandingDisbursement.id) {
       throw new UnprocessableEntityException(
-        "Confirmation of Enrollment and application status update to completed is failed",
+        `${FIRST_COE_NOT_COMPLETE} ${FIRST_COE_NOT_COMPLETE_MESSAGE}`,
       );
     }
 
+    this.disbursementScheduleService.updateDisbursementAndApplicationCOEApproval(
+      disbursementScheduleId,
+      userToken.userId,
+      disbursementSchedule.application.id,
+      disbursementSchedule.application.applicationStatus,
+    );
+
     // Send a message to allow the workflow to proceed.
-    await this.workflow.sendConfirmCOEMessage(application.assessmentWorkflowId);
+    await this.workflow.sendConfirmCOEMessage(
```

Comment:
Do we need send this message for the second COE? I believe that only for the first one right?

Reply:
Yes, I missed to add the condition to check application status.

Reply:
Application status check added.

Reply:
The comments are really helpful ❤️ 
![image](https://user-images.githubusercontent.com/61259237/153296181-c9171ab9-f845-4d1f-96e6-e76ec36e638d.png)